backend/layers/processing/upload_failures/app.py

```diff
@@ -159,12 +159,13 @@ def trigger_slack_notification(
         notify_slack(data, webhook)
 
 
-def cleanup_artifacts(dataset_id: str) -> None:
+def cleanup_artifacts(dataset_id: str, error_step_name: Optional[str] = None) -> None:
     """Clean up artifacts"""
     object_key = os.path.join(os.environ.get("REMOTE_DEV_PREFIX", ""), dataset_id).strip("/")
-    with logger.LogSuppressed(Exception, message="Failed to clean up artifacts."):
-        artifact_bucket = os.environ["ARTIFACT_BUCKET"]
-        delete_many_from_s3(artifact_bucket, object_key + "/")
+    if not error_step_name or error_step_name == "download-validate":
```

Comment:
I assume we aren't too worried about subsequent attempts to upload, which would overwrite these files, while you're in the middle of troubleshooting. Solving for that would probably be over-engineering at this point.

Reply:
in case this is worth considering (now or in the future): on error, h5ad, and/or other data artifacts, could be copied to another location for later troubleshooting, allowing the cleanup step to remain oblivious.

Reply:
@danieljhegeman it's something we can do in the future but not needed for this PR.

Reply:
Proposal: [Dataset Submission Pipeline Error Handling](https://docs.google.com/document/d/1o2vuvl-kXwRJN1nBoPzJS_MAQgDGYnjmPZWa4qRDi-I/edit#heading=h.pxeitln4pjcw)